charts/refarch/README.md

```diff
@@ -0,0 +1,278 @@
+
+This chart provides multiple ways to set configuration for your deployments.
+
+If you have a Spring-based application you can directly configure the content of the application.yml file.
+The content will be placed inside a module-specific ConfigMap which is mounted into the container at runtime.
+You can define environment variables in it as well.
+
+Example:
+```yaml
+  applicationYML:
+    MY_ENV_VARIABLE: someValue
+    spring:
+      profiles:
+        active: "myprofile"
+```
+
+Alternatively you can define single environment variables as follows, optionally referencing and entry inside a global secret or ConfigMap.
+This can e.g. be used to set `JAVA_OPTS_APPEND`.
+
+Example:
+```yaml
+  env:
+    # static
+    - name: MY_ENV_VARIABLE
+      value: myvalue
+    # reference to secret
+    - name: MY_ENV_FROM_SECRET
+      valueFrom:
+        secretKeyRef:
+          name: my-global-secret
+          key: my-key
+    # reference to configmap
+    - name: MY_ENV_FROM_CONFIGMAP
+      valueFrom:
+        configMapKeyRef:
+          name: my-global-configmap
+          key: my-key
+```
+
+If you need to set multiple environment variables at once, you can load the whole content of a global secret or configmap.
+In this case, the name of the keys in the secret or configmap define the name of the created environment variable.
+
+Example:
+```yaml
+  envFrom:
+    # whole files of envs
+    - configMapRef:
+      name: my-global-configmap
+    - secretRef:
+      name: my-global-secret
+```
+
+#### Networking
+
+To expose your deployments (either cluster-interally or externally) services and ingresses can be used.
+
+For services, you can either define all port mappings manually or use the custom defined properties, which configure mappings that are required by most services.
+Setting `http` to `true` maps to:
+```yaml
+    - name: http
+      protocol: TCP
+      port: 8080
+      targetPort: http
+```
+
+Setting `javaDebug` to `true` maps to:
+```yaml
+    - name: debug
+      protocol: TCP
+      port: 8778
+      targetPort: http
+```
+
+Example for configuring the service:
+```yaml
+  service:
+    http: true
+    javaDebug: false
+    ports:
+      - name: custom-port
+        protocol: TCP
+        port: 1234
+        targetPort: 5678
+```
+
+Ingress is required if you want to expose the deployment externally.
+Example:
+```yaml
+  ingress:
+    port: 1234    # defaults to 8080
+    className: "" # optional 
+    annotations: # optional custom annotations
+      kubernetes.io/ingress.class: nginx
+      kubernetes.io/tls-acme: "true"
+    hosts:
+      - host: my-example.local
+        paths:
+          - path: /
+            pathType: ImplementationSpecific
+    tls:
+      - secretName: my-tls-secret
+        hosts:
+          - my-example.local
+```
+
+#### Storage
+
+By default, no volumes are defined and no mounts are mounted connected to the containers. This can be useful to mount important files like certificates.
```

Comment:
```suggestion
By default, no volumes are defined and no mounts are mounted to the containers. This can be useful to attach files like certificates.
```